src/sources/kafka.rs

```diff
@@ -113,16 +117,20 @@ fn kafka_source(
 
                 async move {
                     match message {
-                        Err(e) => {
-                            Err(error!(message = "Error reading message from Kafka", error = ?e))
+                        Err(error) => {
+                            emit!(KafkaEventFailed { error });
+                            Err(())
                         }
                         Ok(msg) => {
+                            emit!(KafkaEventReceived {
+                                byte_size: msg.payload_len()
+                            });
+
                             let payload = match msg.payload_view::<[u8]>() {
                                 None => return Err(()), // skip messages with empty payload
-                                Some(Err(e)) => {
-                                    return Err(
-                                        error!(message = "Cannot extract payload", error = ?e),
-                                    )
+                                Some(Err(_)) => {
```

Comment:
Since we're not actually trying to convert to any type here, we can probably use `payload` and get rid of the extra step and unit error type.